### PR TITLE
refactoring pd failover

### PR DESCRIPTION
**Pull request by weekface**
- update go.mod, remove `code-generator` and other unused packages
- refactor pd failover
- remove `Replicas` attribute
- copy and modify `generate-groups.sh` from code-generator
- adjust the related unit tests

@tennix @xiaojingchen @gregwebs PTAL

### CODE REVIEW

File: pkg/manager/member/tidb_failover_test.go [hunk 1]
```diff
@@ -294,12 +291,10 @@ func TestFakeTiDBFailoverRecover(t *testing.T) {
...
 				tc.Status.TiDB.FailureMembers = map[string]v1alpha1.TiDBFailureMember{
 					"failover-tidb-0": {
-						PodName:  "failover-tidb-0",
-						Replicas: 2,
+						PodName: "failover-tidb-0",
 					},
 					"failover-tidb-1": {
-						PodName:  "failover-tidb-0",
-						Replicas: 3,
+						PodName: "failover-tidb-0",
```
Comment (xiaojingchen) [inline pkg/manager/member/tidb_failover_test.go]:
```suggestion
						PodName: "failover-tidb-1",
```

File: pkg/manager/member/tidb_failover_test.go [hunk 2]
```diff
@@ -320,12 +315,10 @@ func TestFakeTiDBFailoverRecover(t *testing.T) {
...
```
Comment (xiaojingchen) [inline pkg/manager/member/tidb_failover_test.go]:
```suggestion
						PodName: "failover-tidb-1",
```

File: pkg/manager/member/tidb_failover_test.go (hunk 1 above)
Comment (weekface) [reply pkg/manager/member/tidb_failover_test.go]: Why change to `failover-tidb-1 `?

File: pkg/manager/member/tidb_failover_test.go (hunk 2 above)
Comment (weekface) [reply pkg/manager/member/tidb_failover_test.go]: Why change to `failover-tidb-1 `?

File: pkg/manager/member/pd_failover.go [hunk 3]
```diff
@@ -88,26 +90,62 @@ func (pf *pdFailover) Failover(tc *v1alpha1.TidbCluster) error {
...
+		if err != nil {
+			return err
 		}
+		pvcName := ordinalPVCName(v1alpha1.PDMemberType, controller.PDMemberName(tcName), ordinal)
+		pvc, err := pf.pvcLister.PersistentVolumeClaims(ns).Get(pvcName)
+		if err != nil {
+			return err
+		}
+
+		if tc.Status.PD.FailureMembers == nil {
```
Comment (tennix) [inline pkg/manager/member/pd_failover.go]: The previous code already accesses `tc.Status.PD.FailureMembers[podName]`

File: pkg/manager/member/tidb_failover_test.go [hunk 4]
```diff
@@ -256,12 +255,10 @@ func TestFakeTiDBFailoverRecover(t *testing.T) {
...
```
Comment (xiaojingchen) [inline pkg/manager/member/tidb_failover_test.go]:
```suggestion
						PodName: "failover-tidb-1",
```

**Pull request merged**